File: Carthage/Checkouts/ios-library/KinveyKit/KinveyTests/Event.m
//
//  Event.m
//  Kinvey
//
//  Created by Santosh on 01/10/15.
//  Copyright (c) 2015 GQuotient. All rights reserved.
//

#import "Event.h"

@implementation Event

@synthesize entityId,name,date,location,metadata,response,age;

- (NSDictionary *)hostToKinveyPropertyMapping
{
    return @{
             @"entityId" : KCSEntityKeyId, //the required _id field
             @"name" : @"name",
             @"date" : @"date",
             @"location" : @"location",
             @"age" : @"Age",
             @"response" : @"response",
             @"metadata" : KCSEntityKeyMetadata, //optional _metadata field
             @"descriptionEvent" : @"descriptionEvent"
             };
}

@end

File: Carthage/Checkouts/ios-library/Kinvey/KinveyTests/Event.swift
//
//  Event.swift
//  Kinvey
//
//  Created by Victor Barros on 2016-07-08.
//  Copyright © 2016 Kinvey. All rights reserved.
//

import Kinvey

/// Event.swift - an entity in the 'Events' collection
class Event : Entity {

    var name: String?
    var publishDate: Date?
    var location: String?
    
    override class func collectionName() -> String {
        return "Event"
    }
    
    override func propertyMapping(_ map: Map) {
        super.propertyMapping(map)
        
        name <- ("name", map["name"])
        publishDate <- ("date", map["date"], ISO8601DateTransform())
        location <- ("location", map["location"])
    }
    
}
